### DISCUSSION TITLE
How to make bulk deletes work while adhering to ModelPolicy?

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#2148] [Help] [answered]
Hey,

I'm trying to make bulk deletes work in an application where multiple users can delete only some models, but see them all.

So the issue I am facing is that right now, these users can delete all models as the `handleRecordBulkDeletion` is never called when you override it in a List Page...

I saw @danharrin's comment [here](https://github.com/laravel-filament/filament/pull/1844#issuecomment-1064962052) and tried adding the method to my `ListUsers` Page, to no avail.
```php
class ListUsers extends ListRecords
{

    protected static string $resource = UserResource::class;


    /**
     * Loop through every record to be deleted,
     * to check permissions for delete.
     *
     * @param  \Illuminate\Database\Eloquent\Collection  $records
     */
    protected function handleRecordBulkDeletion(Collection $records): void
    {
        logger("bulk delete:", [$records]);
        $records->each(fn(Model $record) => static::getResource()::canDelete($record) ? $record->delete() : null);
    }
}
```

The logger never executes and all models are deleted, disregarding my `UserPolicy` entirely.
Is there any way how I could solve this?

**Comment by madsem** [comment, marked as answer]
Okay, I was just a little stupid hahaha.
Sorry for the uproar!

My particular problem was:

I created a **custom** delete bulk action in my `UserResource`. When you do that, the `ListRecord` bulk delete is never called, Filament actually has a built-in method for bulk deletes!

So I only had to remove the custom bulk action for `delete`, and it was working like a charm.

Closing this but hope it'll help someone in the future.

**Comment by danharrin** [comment]
You can override the ListRecord's `getDeleteBulkAction()` method and just customise that action instead of creating your own and removing ours

```php
parent::getDeleteBulkAction()->label('New label')->action(fn () => new action)
```
